aidial_adapter_openai/utils/prompt_truncate.py

```diff
@@ -0,0 +1,65 @@
+from typing import Callable, List, Set, Tuple, TypeVar
+
+from aidial_sdk.exceptions import (
+    TruncatePromptSystemAndLastUserError,
+    TruncatePromptSystemError,
+)
+
+_T = TypeVar("_T")
+
+DiscardedMessages = List[int]
+UsedTokens = int
+
+
+def truncate_prompt(
+    message_holders: List[_T],
```

Comment:
Because it's not exactly the *message*. For multi-modal have object, that bounds message with data about pictures.

So I want to explicitly separate `message`, that we operate frequently, and `message_holder` - object, that *contains message*, but not bound by that

That's why argument for this function are also `is_system_message_getter` - because we can add custom logic to extract system message from this object

Reply:
But the whole point of a **generic** algorithm is that it doesn't care about the particular implementation of the message type `_T`. It could be **anything** - it's not important for the logic of the algorithm. The algorithm couldn't tell a difference between an extact message and a message holder. It's all the same for it.

What's important for the person who reads this code is only the fact that `_T` represents a **message** in a most generic sense. Given this undersanding one could immediately realize in what contexts this algo might be used.

Reply:
Renamed to `messages`